explorer/src/components/account/AnchorAccountCard.tsx

```diff
@@ -0,0 +1,170 @@
+import React from "react";
+
+import { Account } from "providers/accounts";
+import { Address } from "components/common/Address";
+import { Program, BorshAccountsCoder } from "@project-serum/anchor";
+import { capitalizeFirstLetter } from "utils/anchor";
+import { ErrorCard } from "components/common/ErrorCard";
+import { ErrorBoundary } from "@sentry/react";
+import ReactJson from "react-json-view";
+
+export function AnchorAccountCard({
+  account,
+  program,
+}: {
+  account: Account;
+  program: Program;
+}) {
+  if (!account.details || !account.details.rawData) {
+    return (
+      <ErrorCard text={"This account is parsed as an SPL-native account"} />
+    );
+  }
+  const accountBuffer = account.details.rawData;
+
+  const discriminator = accountBuffer.slice(0, 8) ?? undefined;
+  if (!discriminator) {
+    return <ErrorCard text={"Failed to find anchor account discriminator"} />;
+  }
+
+  let foundAccountLayoutName: string | undefined;
+  let decodedAnchorAccountData: Object | undefined;
+  if (program) {
+    // Iterate all the structs, see if any of the name-hashes match
+    Object.keys(program.account).forEach((accountType) => {
+      const layoutName = capitalizeFirstLetter(accountType);
+      const discriminatorToCheck =
+        BorshAccountsCoder.accountDiscriminator(layoutName);
+
+      if (discriminatorToCheck.equals(discriminator)) {
+        foundAccountLayoutName = layoutName;
+        const accountDecoder = program.account[accountType];
+        decodedAnchorAccountData = accountDecoder.coder.accounts.decode(
+          layoutName,
+          accountBuffer
+        );
+      }
+    });
+  }
+
+  if (!foundAccountLayoutName || !decodedAnchorAccountData) {
+    return (
+      <ErrorCard
+        text={
+          "Failed to find matching anchor account type for account discriminator"
+        }
+      />
+    );
+  }
+
+  return (
+    <ErrorBoundary
+      fallback={<ErrorCard text="Unable to decode anchor account data" />}
+    >
+      <div className="card">
+        <div className="card-header">
+          <div className="row align-items-center">
+            <div className="col">
+              <h3 className="card-header-title">{foundAccountLayoutName}</h3>
+            </div>
+          </div>
+        </div>
+
+        <div className="table-responsive mb-0">
+          <table className="table table-sm table-nowrap card-table">
+            <thead>
+              <tr>
+                <th className="w-1 text-muted">Key</th>
+                <th className="text-muted">Value</th>
+              </tr>
+            </thead>
+            <tbody className="list">
+              {decodedAnchorAccountData &&
+                Object.keys(decodedAnchorAccountData).map((key) => {
+                  // @ts-ignore
+                  return renderAccountRow(key, decodedAnchorAccountData[key]);
+                })}
```

Comment:
Since this is _fundamentally_ a React component, it's _surprisingly_ important to express it as JSX. Rename this to `AccountRow` and do this…

```suggestion
                Object.keys(decodedAnchorAccountData).map((key) => (
                  <AccountRow
                    key={key}
                    value={decodedAnchorAccountData[key]}
                  />
                ))}
```